doc/developer/design/20220819_bounded_input_reliance.md

```diff
@@ -0,0 +1,164 @@
+# Bounded Input Reliance
+
+**Note**: This design document relies on changes in [this pr], which is expected
+to merge soon.
+
+[this pr]: https://github.com/MaterializeInc/materialize/pull/14355
+
+## Summary
+
+In the [source ingestion pipeline design doc] required Materialize to guarantee _bounded input reliance_.
+This means that sources should not re-read data that has already been read and written to persist.
+This requires that Materialize track progress on each source, and restart sources from a
+known good restart point.
+
+[source ingestion pipeline design doc]: https://github.com/MaterializeInc/materialize/pull/12109/files
+[Epic task]: https://github.com/MaterializeInc/materialize/issues/13534
+
+## Jargon
+
+**Note**: this section uses terminology defined in [the original reclocking design doc].
+
+- _resumption frontier_: A _target gauge_ (whose type is `IntoTime`, or currently, an ordinary `Timestamp`) frontier that is
+safe to restart a specific source at.
```

Comment:
As above, it isn't yet clear what a "source" is. I'd introduce some terminology that distinguishes "things like Kafka or postgres" from "an implementor of the source reader" from "the persisted shard".